### PR TITLE
feat(discord): [OWL-26532861] Implement give command

### PR DESCRIPTION
**Pull request by alvinb**

### CODE REVIEW

File: src/discord/commands/give.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      notEnough: `Sorry, you don't have that many ${CONFIG.CURRENCY.PLURAL} to give. :neutral_face:`,
+      success: `You gave ${recipientData.discord_name} ${amount} ${CONFIG.CURRENCY.PLURAL}.`,
+    };
+
+    if (user.points < 1) {
+      await interaction.reply({ content: replies.noPoints });
+      return;
+    }
+
+    if (isNaN(amount)) {
```
Comment (devthena) [inline src/discord/commands/give.ts:50]: Do we still need this block if we are already using `.addNumberOption` for the `amount` input?

File: src/discord/commands/give.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      await interaction.reply({ content: replies.invalidAmount });
+      return;
+    }
+
+    if (amount <= 0) {
```
Comment (devthena) [inline src/discord/commands/give.ts:55]: Just me being nit-picky, can we make the condition `amount < 1` lol

File: src/discord/commands/give.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      await interaction.reply({ content: replies.invalidLowerBound });
+      return;
+    }
+
+    if (user.points < amount) {
+      await interaction.reply({ content: replies.notEnough });
+      return;
+    }
+
+    // @to-do: check if recipient goes will exceed max points before giving
```
Comment (devthena) [inline src/discord/commands/give.ts:65]: I don't think we plan to add a max point for users so we don't have to do this check

**Review by devthena — CHANGES_REQUESTED** [review]

File: src/discord/commands/give.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    if (!CONFIG.GIVE.ENABLED) {
+      await interaction.reply({
+        content: 'Giving points is not enabled in this server.',
+        ephemeral: true,
+      });
+      return;
+    }
+
+    const amount = Number(interaction.options.get('amount')?.value) || 0;
+    const replies = {
```
Comment (devthena) [inline src/discord/commands/give.ts:36]: Let's make it so all replies that are not `success` will be returned as ephemeral on the `interaction.reply`

**Comment by devthena** [comment]
Sorry, one last thing, can you make it so PR title is capitalized please?

File: src/discord/commands/give.ts [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+      option
+        .setName('amount')
+        .setDescription('Enter a specific amount to give')
+        .setRequired(true)
+    ),
+  execute: async (
+    Bots: BotsProps,
+    interaction: CommandInteraction,
+    user: DiscordUserProps,
+    recipientData: DiscordUserProps
```
[the remark this answers is not in the document]
Comment (alvinb) [reply src/discord/commands/give.ts]: addressed here: https://github.com/devthena/little-owl/pull/20/commits/856a4fcca3615a320a231425c094ac5109b63840

File: src/discord/commands/give.ts (hunk 1 above)
Comment (alvinb) [reply src/discord/commands/give.ts]: removed. addressed here: https://github.com/devthena/little-owl/pull/20/commits/856a4fcca3615a320a231425c094ac5109b63840

File: src/discord/commands/give.ts (hunk 3 above)
Comment (alvinb) [reply src/discord/commands/give.ts]: removed here: https://github.com/devthena/little-owl/pull/20/commits/856a4fcca3615a320a231425c094ac5109b63840

File: src/discord/commands/give.ts (hunk 2 above)
Comment (alvinb) [reply src/discord/commands/give.ts]: updated here: https://github.com/devthena/little-owl/pull/20/commits/856a4fcca3615a320a231425c094ac5109b63840

File: src/discord/events/interactionCreate.ts [hunk 6]
```diff
@@ -1,6 +1,6 @@
 import { CommandInteraction } from 'discord.js';
 import { BotsProps, DiscordUserProps } from 'src/interfaces';
-import { CoinFlip, EightBall, Gamble, Help, Points } from '../commands';
+import { CoinFlip, EightBall, Gamble, Help, Points, Give } from '../commands';
```
[the remark this answers is not in the document]
Comment (alvinb) [reply src/discord/events/interactionCreate.ts]: updated here: https://github.com/devthena/little-owl/pull/20/commits/856a4fcca3615a320a231425c094ac5109b63840

File: src/discord/commands/give.ts (hunk 4 above)
Comment (alvinb) [reply src/discord/commands/give.ts:36]: addressed here: https://github.com/devthena/little-owl/pull/20/commits/856a4fcca3615a320a231425c094ac5109b63840

File: src/discord/events/interactionCreate.ts [hunk 7]
```diff
@@ -1,6 +1,14 @@
 import { CommandInteraction } from 'discord.js';
 import { BotsProps, DiscordUserProps } from 'src/interfaces';
-import { CoinFlip, EightBall, Gamble, Help, Points, Star } from '../commands';
+import {
+  CoinFlip,
+  EightBall,
+  Give,
```
Comment (devthena) [inline src/discord/events/interactionCreate.ts:6]: Let's move this line after `Gamble` to make it alphabetical

Comment (alvinb) [reply src/discord/events/interactionCreate.ts]: done

**Review by devthena — APPROVED** [review]

**Pull request merged**